src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -647,42 +648,182 @@ public void Fault(RequestChannel channel)
 
-                        this.TryCompleteHttpRequest(_httpRequestMessage);
-                        return replyMessage;
+                                _httpRequestMessage.Headers.Referrer = new Uri(value);
+                            }
+                            else if (string.Compare(name, "transfer-encoding", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                if (value.ToUpperInvariant().IndexOf("CHUNKED", StringComparison.OrdinalIgnoreCase) != -1)
+                                {
+                                    _httpRequestMessage.Headers.TransferEncodingChunked = true;
+                                }
+                                else
+                                {
+                                    _httpRequestMessage.Headers.TransferEncoding.TryParseAdd(value);
+                                }
+                            }
+                            else if (string.Compare(name, "user-agent", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                _httpRequestMessage.Headers.UserAgent.Add(ProductInfoHeaderValue.Parse(value));
+                            }
+                            else if (string.Compare(name, "if-modified-since", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                DateTimeOffset modifiedSinceDate;
+                                if (DateTimeOffset.TryParse(value, DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AllowWhiteSpaces | DateTimeStyles.AssumeLocal, out modifiedSinceDate))
+                                {
+                                    _httpRequestMessage.Headers.IfModifiedSince = modifiedSinceDate;
+                                }
+                                else
+                                {
+                                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(
+                                        new ProtocolException(SR.Format(SR.HttpIfModifiedSinceParseError, value)));
+                                }
+                            }
+                            else if (string.Compare(name, "date", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                // this will be taken care of by System.Net when we make the request
+                            }
+                            else if (string.Compare(name, "proxy-connection", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                throw ExceptionHelper.PlatformNotSupported("proxy-connection");
+                            }
+                            else if (string.Compare(name, "range", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                // we don't support ranges in v1.
```

Comment:
Question: if we don't support it, why don't we throw PNSE like we do for proxy-connection?
Is there a reason we don't support it?  If so, a comment would be good.
Also, nit -- I'd avoid fictitious versions like 'v1' and just say we don't currently support it.